src/backend/distributed/connection/connection_management.c

```diff
@@ -1478,7 +1516,8 @@ ShouldShutdownConnection(MultiConnection *connection, const int cachedConnection
 		   connection->requiresReplication ||
 		   (MaxCachedConnectionLifetime >= 0 &&
 			MillisecondsToTimeout(connection->connectionEstablishmentStart,
-								  MaxCachedConnectionLifetime) <= 0);
+								  MaxCachedConnectionLifetime) <= 0) ||
+		   RemoteConnectionClosed(connection);
```

Comment:
function comment should be updated accordingly.

Reply:
I just decided to remove this part :) But, some other comments should change, I agree.